src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java21/ExpectedStringTemplateBasicWithTabs.txt

```diff
@@ -377,6 +359,8 @@ COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
         |   |               |           |           `--RPAREN -> ) [46:22]
         |   |               |           `--RPAREN -> ) [46:23]
         |   |               |--EMBEDDED_EXPRESSION_END -> } [47:3]
+        |   |               |--STRING_TEMPLATE_CONTENT ->  there  [47:4]
```

Comment:
I assume there is no easy way to prove we are not missing other text?
Note: There is no removal of `there`, so it seems this was completely gone from our AST.

Reply:
We could, it would be involved though. We could have some verification that all tokenized text exists in the AST.

1. Duplicate the token stream
2. Generate the tree (in code, not text, a 'la DetailAST)
3. Check to make sure that everything that is tokenized exists in the tree via n-nary search

The main reason that we were missing stuff here was due to the weird building of the AST that I used previously to try to keep the lexer as context free as possible.